uiautomationutilities/perf_tool/readme.md

```diff
@@ -1,155 +1,157 @@
+	- If we are having more than one iterations of same scenario, i.e. running same code multiple times, 
+		- Make sure to clear the previous run's markers in the start of the iteration as follows: `CodeMarkerManager.getInstance().clearMarkers();`
+		- Also write the content to the file using [FileAppender](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/logging/appender/FileAppender.java) such as [here](https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/dev/msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/perf/TestCasePerf.java#L119-L124)
+	- Also add followings after performing the particular scenario:
+		- Clear the codemarkers: `CodeMarkerManager.getInstance().clearMarkers();`
+		- Disable the codemarkers: `CodeMarkerManager.getInstance().setEnableCodeMarker(false);`
+- In the C# tool code, open the file [PerfDataConfiguration.xml](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/uiautomationutilities/perf_tool/PerfDataConfiguration.xml) and perform following actions:
+	- In tag `MeasurementsConfigurations`, add a new `MeasurementsConfiguration` with giving the startMarker and endMarker as given in the step 2. 
+	- In tag, `Scenarios`, add a new scenario with the Measurement ID of the MesurementConfiguration just added in previous sub-step.
+- Push the common code and msal native code to dev branch by raising PRs.
+- Push the C# changes of its repo (i.e. common) into dev branch by raising a PR.
+- Deploy C# tool into the pipeline. (Pipeline information given at the start of this doc). We can deploy C# tool even before merging of PR if need be. As this deployment is done after building at the local machine.
 
 
 ### Steps to deploy latest version of C# tool to pipeline:
-- 1. Take a checkout of C# tool repo (Details above)
-- 2. Open the Location of the perf tool (Location details given above) in Visual studio.
-- 3. Build Solution by clicking on the Build -> Build Solution. Make sure Release build is chosen for deployment.
-- 4. Bundle following files into one zip file named "PerfRunnables.zip":
-	- a. Files "PerfIdentity.exe", "PerfIdentity.runtimeconfig.json", "PerfIdentity.dll" and "PerfIdentity.deps.json" from location  "bin\Release\netcoreapp3.1" and file "PerfDataConfiguration.xml" from root location of the project
-- 5. Go to the pipeline and then from the left side options, Open Library page and upload the file "PerfRunnables.zip" after deleting previous one (if exists)
-- 6. Go to the pipeline definition and make change in the task "Download Perf runnable package" by selecting the file name which was uploaded in the "secure file" drop down.
-- 7. Save the build definition. 
+- Install a recent version of [Visual Studio](https://visualstudio.microsoft.com/)
```

Comment:
Probably should call out that Windows is requisite for this. Or is it not?

Reply:
I believe you can also run Visual Studio on Mac